client/state/comments/models/comment.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Internal dependencies
+ */
+import { Author } from './comment-author'; // eslint-disable-line no-unused-vars
+import { CommentInfo } from './comment-info'; // eslint-disable-line no-unused-vars
+
+export function Comment( constructor ) {
+	Object.defineProperty( this, 'constructor', {
+		value: constructor || this
+	} );
+}
+
+export function LoadingComment( siteId, commentId ) {
+	const comment = new Comment( LoadingComment );
+
+	comment.siteId = siteId;
+	comment.commentId = commentId;
+
+	return comment;
+}
+
+/**
+ * Create a new loaded comment value
+ *
+ * @param {Number} siteId site id
+ * @param {Number} postId post id
+ * @param {Number} commentId comment id
+ * @param {Author} author comment author
+ * @param {CommentInfo} info comment information
+ * @returns {Comment} represents a comment with its content
+ * @constructor
+ */
+export function LoadedComment( {
```

Comment:
maybe. I needed/wanted a top-level type. maybe it's habit, maybe it's good practice. the `Comment` itself could be renamed to `BaseComment` but I like being able to express that a component requires a `Comment` and it doesn't care if it's loaded or not.